stock_move_backdating/models/stock_quant.py

```diff
@@ -0,0 +1,20 @@
+#  Copyright 2023 Simone Rubino - TAKOBI
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockQuant (models.Model):
+    _inherit = 'stock.quant'
+
+    def _update_available_quantity(
+        self, product_id, location_id, quantity,
+        lot_id=None, package_id=None, owner_id=None, in_date=None,
+    ):
+        date_backdating = self.env.context.get('date_backdating')
+        if date_backdating is not None and in_date is not None:
```

Comment:
Why if `in_date is not None`?

Reply:
Because in `10.0` only the dates of `quant_ids` were updated and the field `quant_ids` only contains the quants in the destination location:
![image](https://user-images.githubusercontent.com/101628000/212642541-f4545708-2ab2-4abf-ba0e-8661f1333505.png)
(this is the move of a delivery)

that is why, seeing how this method is called in https://github.com/odoo/odoo/blob/3f4654f9cb0a05c65195745c78f494c228e09549/addons/stock/models/stock_move_line.py#L442-L450, I only wanted to change the date of the quant in the destination (last line, with parameter `in_date`) but not the date of the quant in the source location (first line, without parameter `in_date`).

On a second thought, it might make more sense to also change the quant in the source location so that the dates stay aligned; I added this change and its test in their own commit.